spring-vault-core/src/main/java/org/springframework/vault/core/VaultTransitTemplate.java

```diff
@@ -340,6 +343,61 @@ public String rewrap(String keyName, String ciphertext,
 				.getRequiredData().get("ciphertext");
 	}
 
+	@Override
+	public String generateHmac(String keyName, String plainText) {
+		return generateHmac(keyName, plainText, DEFAULT_SIGN_ALGORITHM);
+	}
+
+	@Override
+	public String generateHmac(String keyName, String plainText, String algorithm) {
+		Assert.hasText(keyName, "KeyName must not be empty");
+		Assert.notNull(plainText, "Plain text must not be null");
+
+		Map<String, String> request = new LinkedHashMap<>();
+
+		request.put("input", Base64Utils.encodeToString(plainText.getBytes()));
+
+		return (String) vaultOperations.
+				write(String.format("%s/hmac/%s/%s", path, keyName, algorithm), request).getData()
```

Comment:
Let's not pass the algorithm as path variable but as part of the body. Key types like ed25519 have their own hashing algorithms and a path-based variant does not allow using the key-specifics.